File: Python/Baekjoon/15649.py
import sys 
input = sys.stdin.readline

N, M = map(int, input().split())
visited = [False for _ in range(N+1)]
stack = []

def dfs():
    if len(stack) == M:
        print(*stack) 
        return 

    for number in range(1,N+1): 
        if not visited[number]:
            visited[number] = True
            stack.append(number)
            dfs()
            visited[number] = False 
            stack.pop() 
            

dfs()

File: JavaScript/Baekjoon/15649.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

function solution (input) {
    [n, m] = input[0].split(' ')
    const N = Number(n)
    const M = Number(m)

    let result = ''
    const output = []
    const visited = new Array(N).fill(false)


    function dfs (count) {
        if (count === M ) {
            result += `${output.join(' ')}\n`
            return
        }

        for (let i = 0 ;i < N; i++){
            if(visited[i] === true) continue;
            visited[i] = true
            output.push(i+1)
            console.log("dd",output)
            console.log(visited)
            dfs(count + 1)
            output.pop()
            visited[i] = false

        }
    }
    dfs(0)
    return result
}

console.log(solution(input))